### PR TITLE
fix(replacements) Add even more metrics to troubleshoot the PostReplacementConsistencyEnforcer

### PR DESCRIPTION
**Pull request by fpacifici**
PR #932 (https://github.com/getsentry/snuba/pull/932/files#diff-249a3fa811e95624b5a47320cc575abaR130) showed some discrepancies.
The problem is that, it seems the local variable picked up when creating the sentry event are capped and not all the local variables are visible in the stack trace. Critically missing the ones I need: the original condition from the query.

This tweaks the warning to ensure the groups that are not matching are highlighted properly.

### CODE REVIEW

File: snuba/datasets/storages/processors/replaced_groups.py [hunk 1]
```diff
@@ -124,12 +125,11 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
...
-                log_issue = True
-
-            if log_issue:
                 logger.warning(
-                    "Discrepancy between project_extension and replacement consistency processor",
+                    "Groups discrepancy between project_extension and processor.",
+                    extra={
+                        "extension": existing_groups_conditions,
+                        "processor": condition_to_add,
+                    },
```
Comment (tkaemming) [inline snuba/datasets/storages/processors/replaced_groups.py:133]: Not sure if this goes to Kibana, but @JTCunning yelled at me and Dan for putting non-scalars in logging yesterday.

**Review by tkaemming — APPROVED** [review]
This is fine by me, probably check with @JTCunning for reason noted inline before merging/deploying just to be careful.

File: snuba/datasets/storages/processors/replaced_groups.py (hunk 1 above)
Comment (fpacifici) [reply snuba/datasets/storages/processors/replaced_groups.py:133]: @JTCunning ? 
I can easily turn them into strings.

Comment (JTCunning) [reply snuba/datasets/storages/processors/replaced_groups.py:133]: You're good, here. This goes into StackDriver Logging. I'd actually like to see how they handle some of these more complex non-scalar values. It's only the main Sentry product that goes into the EFK stack.

**Pull request merged**